code/game/objects/items/weapons/scrolls.dm

```diff
@@ -16,92 +16,59 @@
 	uses = 1
 	origin_tech = "bluespace=5"
 
-/obj/item/teleportation_scroll/attack_self(mob/user as mob)
-	user.set_machine(src)
-	var/dat = "<B>Teleportation Scroll:</B><BR>"
-	dat += "Number of uses: [src.uses]<BR>"
-	dat += "<HR>"
-	dat += "<B>Four uses use them wisely:</B><BR>"
-	dat += "<A href='byond://?src=[UID()];spell_teleport=1'>Teleport</A><BR>"
-	dat += "Kind regards,<br>Wizards Federation<br><br>P.S. Don't forget to bring your gear, you'll need it to cast most spells.<HR>"
-	user << browse(dat, "window=scroll")
-	onclose(user, "scroll")
-	return
+/obj/item/teleportation_scroll/examine(mob/user)
+	. = ..()
+	. += "<span class='notice'>Number of uses: [uses]. This scroll will vanish after the final use.</span>"
+	. += "<span class='notice'>P.S. Don't forget to bring your gear, you'll need it to cast most spells.</span>"
 
-/obj/item/teleportation_scroll/Topic(href, href_list)
-	..()
-	if(usr.stat || usr.restrained() || src.loc != usr)
+/obj/item/teleportation_scroll/attack_self(mob/user)
+	if(!uses) //somehow?
+		qdel(src)
```

Comment:
```suggestion
		to_chat(user, "<span class='warning'>You attempt to read the scroll but it disintegrates in your hand, it appears that is has run out of charges!</span>")
		qdel(src)
```
perhaps some feedback would be good to the player here, if they somehow get a bugged out scroll, it'd probably be better if they know why it just poofs out of existence in their hand rather than just QDEL'ing it and leaving them wondering.

Reply:
I was contemplating this, but didn't want to just copy-paste the to_chat from below. This works.